src/HasMedia/HasMediaTrait.php

```diff
@@ -129,10 +129,15 @@ public function addMediaFromUrl(string $url, ...$allowedMimeTypes)
         $this->guardAgainstInvalidMimeType($tmpFile, $allowedMimeTypes);
 
         $filename = basename(parse_url($url, PHP_URL_PATH));
-        if ($filename == '') {
-            $mediaExt = explode('/', mime_content_type($tmpFile));
 
-            $filename = 'file.'.$mediaExt[1];
+        $mediaExt = explode('/', mime_content_type($tmpFile));
+
+        if ($filename === '') {
+            $filename = 'file';
+        }
+
+        if (! str_contains($filename, '.')) {
+            $filename = "{$filename}.{$mediaExt[1]}";
```

Comment:
Are we sure `$mediaExt` will always be an array with at least two elements? Otherwise I'd add an `isset`-like check and a proper fallback.

Reply:
Yes I just checked and there is always a slash in a mime type and it is always present. I can add the check but it is not necessary to do so.

Reply:
I can imagine there could be a corrupted image which mime type doesn't contain the slash? I would personally handle it like so.

```php
$extension = $mediaExt[1] ?? null;

if (!$extension) {
    // ...
}
```